# Data Collection

The purpose of this notebook is to demonstrate how the data for the Summary Notebook was collected, in case you are interested in trying to reproduce this study in some fashion.

## Knight Commission

The first source used in this project is the [Knight Commission on Intercollegiate Athletics](https://www.knightcommission.org/). Their database is accessible [here](https://cafidatabase.knightcommission.org/). From the databse website, click on "Go to Custom Reporting."

![img](../Images/Knight_custom_reporting.png)

From there, you can customize and select any variables with which you are interested. Given that I wanted to focus on athletic spending (in particular football spending), I selected all variables for every option available on this website. Note that I selected "All Institutions," but did not select any of the NCAA or FBS variables in the section below. I also checked the box to adjust for cost inflation.

![img](../Images/Knight_data.png)

![img](../Images/Knight_schools.png)

![img](../Images/Knight_ncaa_years.png)

When you click "Generate Report" at the bottom of the page, you are then taken to a page titled "Custom Reporting Results." You can either examine the results directly on the website, or if you want to download the data, you can click "Download Data." Note that you will have to enter some personal information before you can download the excel file.

![img](../Images/Knight_custom_reporting_results.png)

Once downloaded, the Excel spreadsheet contains all of the data selected along with a data dictionary on a separate sheet. The formatting is a little odd when you try to open the original Excel sheet using pandas, given that the first three columns are frozen. You can either unfreeze the columns directly on the sheet, or you can create a new Excel documnet, copy/paste the first three rows, and then copy/paste the rest. There are a lot of missing values in this data set, but we will deal with that later when merging all three of our data sets.

In [1]:
import pandas as pd

In [2]:
knight = pd.read_excel('../Data/knight_05_19.xlsx')
knight.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5295 entries, 0 to 5294
Data columns (total 31 columns):
 #   Column                                                                 Non-Null Count  Dtype  
---  ------                                                                 --------------  -----  
 0   Data                                                                   5295 non-null   object 
 1   IPEDS ID                                                               5295 non-null   int64  
 2   Year                                                                   5295 non-null   int64  
 3   NCAA Subdivision                                                       5295 non-null   object 
 4   FBS Conference                                                         1853 non-null   object 
 5   Total Expenses                                                         3370 non-null   float64
 6   Excess Transfers Back                                                  3370 non-null   f

---

## Equity in Athletics Data Analysis (EADA)

The next source used in this project is the [Equity in Athletics Data Analysis website](https://ope.ed.gov/athletics/#/). From the homepage, click on "Download Custom Data." On this page you can choose any number of institutions you would like, but for my project I clicked "Select All" so that I could get data on as many colleges as possible.

![img](../Images/EADA_institutions.png)

On the next page, I selected years between 2005–2019 to be consistent with the Knight Commission data, but as you can see they also have data from 2003–2003 available. 

![img](../Images/EADA_years.png)

There are a lot of options for downloading data on this website, but for the purposes of my project this is what I selected:
- Athletics Participation - All Sports and Men's, Women's and Coed Teams
    - Contains the number of student athletes at each institution
- Head Coaches - Men's Team - All Sports and Head Coaches University Employee
    - Contains the number of head coaches for every sport at each institution. 
- Assistant Coaches - Men's Team - All Sports and Assistant Coaches University Employee
    - Contains the number of assistant coaches for every sport at each institution.
- Revenues - All Sports and Men's, Women's and Coed Teams
    - Contains the amount of revenue for all teams at each institution.
- Expenses - All Sports and Men's, Women's and Coed Teams
    - Contains the amount of expenses for all teams at each institution.

![img](../Images/EADA_numbers.png)

![img](../Images/EADA_expenses.png)

If you select multiple options like I did, when you click "Download" at the bottom of the page you will download a ZIP file that contains a CSV file for each option you selected. I will go through each of these five files here in turn so you can see how I got to the final version of the file in my summary notebook.

### Student Athletes
We can see this file contains 209 columns with information about participation for all athletics programs at each institution. However, for my project I was only concerned with the total number of student athletes. Thus, I created a new dataframe that just has 'Unduplicated Count Men's Participation' and 'Unduplicated Count Women's Participation', along with the first four columns containing information about the year and school.

In [3]:
stu_ath = pd.read_csv('../Data/EADA_stud_ath_all_years.csv')

In [4]:
stu_ath.head()

Survey Year  UNITID                      Institution Name State CD  \
0         2005  222178          Abilene Christian University       TX   
1         2005  138558  Abraham Baldwin Agricultural College       GA   
2         2005  126182                   Adams State College       CO   
3         2005  188429                    Adelphi University       NY   
4         2005  188438          Adirondack Community College       NY   

                   Classification Name Classification Other  Sanction Code  \
0     NCAA Division II (with football)                  NaN              1   
1                     NJCAA Division I                  NaN              3   
2     NCAA Division II (with football)                  NaN              1   
3  NCAA Division II (without football)                  NaN              1   
4                   NJCAA Division III                  NaN              3   

  Sanction Name  Male Undergraduates  Female Undergraduates  ...  \
0          NCAA               1792.0                 2137.0  ...   
1         NJCAA               1103.0                 1134.0  ...   
2          NCAA                851.0                  929.0  ...   
3          NCAA               1155.0                 2806.0  ...   
4         NJCAA                874.0                 1141.0  ...   

   Other Sports Coed Team Women Participation  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   

   Other Sports Total Participation  Total Men's Team Participation  \
0                               NaN                           207.0   
1                               NaN                            58.0   
2                               NaN                           272.0   
3                               NaN                           156.0   
4                               NaN                            62.0   

   Total Women's Team Participation  Total Coed Team Men Participation  \
0                              89.0                                NaN   
1                              40.0                                NaN   
2                             114.0                                NaN   
3                             137.0                                NaN   
4                              34.0                                NaN   

   Total Coed Team Women Participation  Grand Total Men's Participation  \
0                                  NaN                              207   
1                                  NaN                               58   
2                                  NaN                              272   
3                                  NaN                              156   
4                                  NaN                               62   

   Grand Total Women's Participation  Unduplicated Count Men's Participation  \
0                                 89                                   207.0   
1                                 40                                    58.0   
2                                114                                   264.0   
3                                137                                   145.0   
4                                 34                                    62.0   

   Unduplicated Count Women's Participation  
0                                      89.0  
1                                      40.0  
2                                     110.0  
3                                     105.0  
4                                      30.0  

[5 rows x 209 columns]

In [5]:
stu_ath = stu_ath[['Survey Year', 'UNITID', 'Institution Name', 'Classification Name', 'Unduplicated Count Men\'s Participation', 'Unduplicated Count Women\'s Participation']]
                   

I then renamed the last two columns so that I would not have to deal with apostrophes.

In [6]:
stu_ath.rename(columns={'Unduplicated Count Men\'s Participation': 'Total Male Participation',
                       'Unduplicated Count Women\'s Participation': 'Total Female Participation'},
              inplace=True)

stu_ath

Survey Year  UNITID                      Institution Name  \
0             2005  222178          Abilene Christian University   
1             2005  138558  Abraham Baldwin Agricultural College   
2             2005  126182                   Adams State College   
3             2005  188429                    Adelphi University   
4             2005  188438          Adirondack Community College   
...            ...     ...                                   ...   
29441         2019  181853                          York College   
29442         2019  217059          York College of Pennsylvania   
29443         2019  141361                  Young Harris College   
29444         2019  206695           Youngstown State University   
29445         2019  126119                          Yuba College   

                       Classification Name  Total Male Participation  \
0         NCAA Division II (with football)                     207.0   
1                         NJCAA Division I                      58.0   
2         NCAA Division II (with football)                     264.0   
3      NCAA Division II (without football)                     145.0   
4                       NJCAA Division III                      62.0   
...                                    ...                       ...   
29441                     NAIA Division II                     166.0   
29442   NCAA Division III without football                     259.0   
29443    NCAA Division II without football                     191.0   
29444                  NCAA Division I-FCS                     260.0   
29445                                CCCAA                     134.0   

       Total Female Participation  
0                            89.0  
1                            40.0  
2                           110.0  
3                           105.0  
4                            30.0  
...                           ...  
29441                       146.0  
29442                       221.0  
29443                       123.0  
29444                       187.0  
29445                        55.0  

[29446 rows x 6 columns]

### Head Coaches
We can see this file contains 207 columns with information about head coaches for all athletics programs at each institution. However, for my project I was only concerned with the total number of head football coaches. Thus, I created a new dataframe that just has 'Football Total', along with the first four columns containing information about the year and school. I determined that this was the column I needed by creating a list comprehension to find columns that had 'Football' in the name, and then selected the one column that represents the total number of head football coaches at each institution.

In [7]:
hc = pd.read_csv('../Data/EADA_head_coaches_all_years.csv')
hc.head()

Survey Year  UNITID                      Institution Name State CD  \
0         2005  222178          Abilene Christian University       TX   
1         2005  138558  Abraham Baldwin Agricultural College       GA   
2         2005  126182                   Adams State College       CO   
3         2005  188429                    Adelphi University       NY   
4         2005  188438          Adirondack Community College       NY   

                   Classification Name Classification Other  Sanction Code  \
0     NCAA Division II (with football)                  NaN              1   
1                     NJCAA Division I                  NaN              3   
2     NCAA Division II (with football)                  NaN              1   
3  NCAA Division II (without football)                  NaN              1   
4                   NJCAA Division III                  NaN              3   

  Sanction Name  Male Undergraduates  Female Undergraduates  ...  \
0          NCAA               1792.0                 2137.0  ...   
1         NJCAA               1103.0                 1134.0  ...   
2          NCAA                851.0                  929.0  ...   
3          NCAA               1155.0                 2806.0  ...   
4         NJCAA                874.0                 1141.0  ...   

   Other Sports Male HD Coach FT Univ. Emp.  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       NaN   

  Other Sports Male HD Coach PT Univ. Emp.  \
0                                      NaN   
1                                      NaN   
2                                      NaN   
3                                      NaN   
4                                      NaN   

   Other Sports Female HD Coach FT Univ. Emp.  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   

   Other Sports Female HD Coach PT Univ. Emp.  Other Sports Total  \
0                                         NaN                 NaN   
1                                         NaN                 NaN   
2                                         NaN                 NaN   
3                                         NaN                 NaN   
4                                         NaN                 NaN   

   Total Male HD Coach FT Univ. Emp.  Total Male HD Coach PT Univ. Emp.  \
0                                  4                                  3   
1                                  3                                  1   
2                                  4                                  1   
3                                  5                                  3   
4                                  3                                  2   

   Total Female HD Coach FT Univ. Emp.  Total Female HD Coach PT Univ. Emp.  \
0                                    0                                    0   
1                                    0                                    0   
2                                    0                                    0   
3                                    0                                    1   
4                                    0                                    0   

   Total HD Coaches  
0                 7  
1                 4  
2                 5  
3                 9  
4                 5  

[5 rows x 207 columns]

In [8]:
hc_fb_cols = [col for col in hc.columns if 'Football' in col]
hc_fb_cols

['Football Male HD Coach FT Univ. Emp.',
 'Football Male HD Coach PT Univ. Emp.',
 'Football Female HD Coach FT Univ. Emp.',
 'Football Female HD Coach PT Univ. Emp.',
 'Football Total']

In [9]:
hc = hc[['Survey Year', 'UNITID', 'Institution Name', 'Classification Name', 'Football Total']]

I then renamed this column 'HC Total' to be more descriptive.

In [10]:
hc.rename(columns={'Football Total': 'HC Total'},
         inplace=True)

hc

Survey Year  UNITID                      Institution Name  \
0             2005  222178          Abilene Christian University   
1             2005  138558  Abraham Baldwin Agricultural College   
2             2005  126182                   Adams State College   
3             2005  188429                    Adelphi University   
4             2005  188438          Adirondack Community College   
...            ...     ...                                   ...   
29207         2019  181853                          York College   
29208         2019  217059          York College of Pennsylvania   
29209         2019  141361                  Young Harris College   
29210         2019  206695           Youngstown State University   
29211         2019  126119                          Yuba College   

                       Classification Name  HC Total  
0         NCAA Division II (with football)       1.0  
1                         NJCAA Division I       NaN  
2         NCAA Division II (with football)       1.0  
3      NCAA Division II (without football)       NaN  
4                       NJCAA Division III       NaN  
...                                    ...       ...  
29207                     NAIA Division II       NaN  
29208   NCAA Division III without football       NaN  
29209    NCAA Division II without football       NaN  
29210                  NCAA Division I-FCS       1.0  
29211                                CCCAA       1.0  

[29212 rows x 5 columns]

### Assistant Coaches
This process was pretty similar to the head coaches steps outlined above. We can see this file contains 207 columns with information about assistant coaches for all athletics programs at each institution. However, for my project I was only concerned with the total number of assistant football coaches. Thus, I created a new dataframe that just has 'Football Total', along with the first four columns containing information about the year and school. I determined that this was the column I needed by creating a list comprehension to find columns that had 'Football' in the name, and then selected the one column that represents the total number of head football coaches at each institution.

In [11]:
ac = pd.read_csv('../Data/EADA_asst_coaches_all_years.csv')
ac.head()

Survey Year  UNITID                      Institution Name State CD  \
0         2005  222178          Abilene Christian University       TX   
1         2005  138558  Abraham Baldwin Agricultural College       GA   
2         2005  126182                   Adams State College       CO   
3         2005  188429                    Adelphi University       NY   
4         2005  188438          Adirondack Community College       NY   

                   Classification Name Classification Other  Sanction Code  \
0     NCAA Division II (with football)                  NaN              1   
1                     NJCAA Division I                  NaN              3   
2     NCAA Division II (with football)                  NaN              1   
3  NCAA Division II (without football)                  NaN              1   
4                   NJCAA Division III                  NaN              3   

  Sanction Name  Male Undergraduates  Female Undergraduates  ...  \
0          NCAA               1792.0                 2137.0  ...   
1         NJCAA               1103.0                 1134.0  ...   
2          NCAA                851.0                  929.0  ...   
3          NCAA               1155.0                 2806.0  ...   
4         NJCAA                874.0                 1141.0  ...   

   Other Sports Male Assistant Coach FT Univ. Emp.  \
0                                              NaN   
1                                              NaN   
2                                              NaN   
3                                              NaN   
4                                              NaN   

  Other Sports Male Assistant Coach PT Univ. Emp.  \
0                                             NaN   
1                                             NaN   
2                                             NaN   
3                                             NaN   
4                                             NaN   

   Other Sports Female Assistant Coach FT Univ. Emp.  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

   Other Sports Female Assistant Coach PT Univ. Emp.  Other Sports Total  \
0                                                NaN                 NaN   
1                                                NaN                 NaN   
2                                                NaN                 NaN   
3                                                NaN                 NaN   
4                                                NaN                 NaN   

   Total Male Assistant Coach FT Univ. Emp.  \
0                                         9   
1                                         0   
2                                         6   
3                                         1   
4                                         0   

   Total Male Assistant Coach PT Univ. Emp.  \
0                                         4   
1                                         1   
2                                         3   
3                                        17   
4                                         3   

   Total Female Assistant Coach FT Univ. Emp.  \
0                                           0   
1                                           0   
2                                           0   
3                                           1   
4                                           0   

   Total Female Assistant Coach PT Univ. Emp.  Total Assistant Coaches  
0                                           1                       14  
1                                           0                        1  
2                                           0                        9  
3                                           2                       21  
4                     

In [12]:
ac_fb_cols = [col for col in ac.columns if 'Football' in col]
ac_fb_cols

['Football Male Assistant Coach FT Univ. Emp.',
 'Football Male Assistant Coach PT Univ. Emp.',
 'Football Female Assistant Coach FT Univ. Emp.',
 'Football Female Assistant Coach PT Univ. Emp.',
 'Football Total']

In [13]:
ac = ac[['Survey Year', 'UNITID', 'Institution Name', 'Classification Name', 'Football Total']]

Like above, I renamed this column 'AC Total' to be more descriptive

In [14]:
ac.rename(columns={'Football Total': 'AC Total'},
         inplace=True)

ac

Survey Year  UNITID                      Institution Name  \
0             2005  222178          Abilene Christian University   
1             2005  138558  Abraham Baldwin Agricultural College   
2             2005  126182                   Adams State College   
3             2005  188429                    Adelphi University   
4             2005  188438          Adirondack Community College   
...            ...     ...                                   ...   
28509         2019  181853                          York College   
28510         2019  217059          York College of Pennsylvania   
28511         2019  141361                  Young Harris College   
28512         2019  206695           Youngstown State University   
28513         2019  126119                          Yuba College   

                       Classification Name  AC Total  
0         NCAA Division II (with football)       7.0  
1                         NJCAA Division I       NaN  
2         NCAA Division II (with football)       6.0  
3      NCAA Division II (without football)       NaN  
4                       NJCAA Division III       NaN  
...                                    ...       ...  
28509                     NAIA Division II       NaN  
28510   NCAA Division III without football       NaN  
28511    NCAA Division II without football       NaN  
28512                  NCAA Division I-FCS      10.0  
28513                                CCCAA       4.0  

[28514 rows x 5 columns]

### Revenues
We can see this file contains 168 columns with information about revenues for all athletics programs at each institution. However, for my project I was only concerned with the total revenue. Thus, I created a new dataframe that just has the last column 'Grand Total Revenue', along with the first four columns containing information about the year and school.

In [15]:
rev = pd.read_csv('../Data/EADA_revenues_all_years.csv')
rev.head()

Survey Year  UNITID                      Institution Name State CD  \
0         2005  222178          Abilene Christian University       TX   
1         2005  138558  Abraham Baldwin Agricultural College       GA   
2         2005  126182                   Adams State College       CO   
3         2005  188429                    Adelphi University       NY   
4         2005  188438          Adirondack Community College       NY   

                   Classification Name Classification Other  Sanction Code  \
0     NCAA Division II (with football)                  NaN              1   
1                     NJCAA Division I                  NaN              3   
2     NCAA Division II (with football)                  NaN              1   
3  NCAA Division II (without football)                  NaN              1   
4                   NJCAA Division III                  NaN              3   

  Sanction Name  Male Undergraduates  Female Undergraduates  ...  \
0          NCAA               1792.0                 2137.0  ...   
1         NJCAA               1103.0                 1134.0  ...   
2          NCAA                851.0                  929.0  ...   
3          NCAA               1155.0                 2806.0  ...   
4         NJCAA                874.0                 1141.0  ...   

   Wrestling Total Revenue  Other Sports Men's Team Revenue  \
0                      NaN                              NaN   
1                      NaN                              NaN   
2                 163368.0                              NaN   
3                      NaN                              NaN   
4                      NaN                              NaN   

   Other Sports Women's Team Revenue  Other Sports Coed Team Revenue  \
0                                NaN                             NaN   
1                                NaN                             NaN   
2                                NaN                             NaN   
3                                NaN                             NaN   
4                                NaN                             NaN   

   Other Sports Total Revenue  Total Men's Team Revenue  \
0                         NaN                 2934983.0   
1                         NaN                  307008.0   
2                         NaN                 1447101.0   
3                         NaN                 1650217.0   
4                         NaN                   12914.0   

   Total Women's Team Revenue  Total Coed Team Revenue  \
0                   1632933.0                      NaN   
1                    282614.0                      NaN   
2                   1080955.0                      NaN   
3                   1407314.0                      NaN   
4                      4940.0                      NaN   

   Not Allocated by Gender/Sport Revenue  Grand Total Revenue  
0                               832469.0              5400385  
1                                53675.0               643297  
2                               253607.0              2781663  
3                              1192266.0              4249797  
4                                96986.0               114840  

[5 rows x 168 columns]

In [16]:
rev = rev[['Survey Year', 'UNITID', 'Institution Name', 'Classification Name', 'Grand Total Revenue']]
rev

Survey Year  UNITID                      Institution Name  \
0             2005  222178          Abilene Christian University   
1             2005  138558  Abraham Baldwin Agricultural College   
2             2005  126182                   Adams State College   
3             2005  188429                    Adelphi University   
4             2005  188438          Adirondack Community College   
...            ...     ...                                   ...   
29441         2019  181853                          York College   
29442         2019  217059          York College of Pennsylvania   
29443         2019  141361                  Young Harris College   
29444         2019  206695           Youngstown State University   
29445         2019  126119                          Yuba College   

                       Classification Name  Grand Total Revenue  
0         NCAA Division II (with football)              5400385  
1                         NJCAA Division I               643297  
2         NCAA Division II (with football)              2781663  
3      NCAA Division II (without football)              4249797  
4                       NJCAA Division III               114840  
...                                    ...                  ...  
29441                     NAIA Division II              3805596  
29442   NCAA Division III without football              2857778  
29443    NCAA Division II without football              5751311  
29444                  NCAA Division I-FCS             16292051  
29445                                CCCAA               743969  

[29446 rows x 5 columns]

### Expenses
This process was similar to the revenues section above. We can see this file contains 168 columns with information about expenses for all athletics programs at each institution. However, for my project I was only concerned with the total expenses. Thus, I created a new dataframe that just has the last column 'Grand Total Expenses', along with the first four columns containing information about the year and school.

In [17]:
exp = pd.read_csv('../Data/EADA_expenses_all_years.csv')
exp

Survey Year  UNITID                      Institution Name State CD  \
0             2005  222178          Abilene Christian University       TX   
1             2005  138558  Abraham Baldwin Agricultural College       GA   
2             2005  126182                   Adams State College       CO   
3             2005  188429                    Adelphi University       NY   
4             2005  188438          Adirondack Community College       NY   
...            ...     ...                                   ...      ...   
29441         2019  181853                          York College       NE   
29442         2019  217059          York College of Pennsylvania       PA   
29443         2019  141361                  Young Harris College       GA   
29444         2019  206695           Youngstown State University       OH   
29445         2019  126119                          Yuba College       CA   

                       Classification Name Classification Other  \
0         NCAA Division II (with football)                  NaN   
1                         NJCAA Division I                  NaN   
2         NCAA Division II (with football)                  NaN   
3      NCAA Division II (without football)                  NaN   
4                       NJCAA Division III                  NaN   
...                                    ...                  ...   
29441                     NAIA Division II                  NaN   
29442   NCAA Division III without football                  NaN   
29443    NCAA Division II without football                  NaN   
29444                  NCAA Division I-FCS                  NaN   
29445                                CCCAA                  NaN   

       Sanction Code Sanction Name  Male Undergraduates  \
0                  1          NCAA               1792.0   
1                  3         NJCAA               1103.0   
2                  1          NCAA                851.0   
3                  1          NCAA               1155.0   
4                  3         NJCAA                874.0   
...              ...           ...                  ...   
29441              2          NAIA                212.0   
29442              1          NCAA               1596.0   
29443              1          NCAA                433.0   
29444              1          NCAA               4048.0   
29445              9         other                917.0   

       Female Undergraduates  ...  Wrestling Total Expenses  \
0                     2137.0  ...                       NaN   
1                     1134.0  ...                       NaN   
2                      929.0  ...                  163368.0   
3                     2806.0  ...                       NaN   
4                     1141.0  ...                       NaN   
...                      ...  ...                       ...   
29441                  191.0  ...                  376715.0   
29442                 2085.0  ...                  107567.0   
29443                  468.0  ...                       NaN   
29444                 4528.0  ...                       NaN   
29445                 1290.0  ...                       NaN   

       Other Sports Men's Team Expenses  Other Sports Women's Team Expenses  \
0                                   NaN                                 NaN   
1                                   NaN                                 NaN   
2                                   NaN                                 NaN   
3                                   NaN                                 NaN   
4                                   NaN                                 NaN   
...                                 ...                                 ...   
29441                               NaN                            107334.0   
29442                               NaN                                 NaN   
29443                               NaN                                 NaN   
29444                

In [18]:
exp = exp[['Survey Year', 'UNITID', 'Institution Name', 'Classification Name', 'Grand Total Expenses']]
exp

Survey Year  UNITID                      Institution Name  \
0             2005  222178          Abilene Christian University   
1             2005  138558  Abraham Baldwin Agricultural College   
2             2005  126182                   Adams State College   
3             2005  188429                    Adelphi University   
4             2005  188438          Adirondack Community College   
...            ...     ...                                   ...   
29441         2019  181853                          York College   
29442         2019  217059          York College of Pennsylvania   
29443         2019  141361                  Young Harris College   
29444         2019  206695           Youngstown State University   
29445         2019  126119                          Yuba College   

                       Classification Name  Grand Total Expenses  
0         NCAA Division II (with football)               5074419  
1                         NJCAA Division I                643297  
2         NCAA Division II (with football)               2781663  
3      NCAA Division II (without football)               3556017  
4                       NJCAA Division III                111670  
...                                    ...                   ...  
29441                     NAIA Division II               3805596  
29442   NCAA Division III without football               2638512  
29443    NCAA Division II without football               5751311  
29444                  NCAA Division I-FCS              16292051  
29445                                CCCAA                743969  

[29446 rows x 5 columns]

### Merging EADA
Now that we have the data needed for this project, all that is left to do is merge all five of these data frames. For each merge, I joined on the 'Survey Year', 'UNITID', 'Institution Name', and 'Classification Name' columns using an outer join to retain as much data as possible. There are still missing values after these outer joins, but I deal with those in the summary notebook so for now I am not concerned.

In [19]:
eada_merge_1 = pd.merge(rev, exp, how = 'outer',
                       left_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'],
                       right_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'])

eada_merge_2 = pd.merge(eada_merge_1, stu_ath, how = 'outer',
                       left_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'],
                       right_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'])

eada_merge_3 = pd.merge(eada_merge_2, hc, how = 'outer',
                       left_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'],
                       right_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'])

eada_final = pd.merge(eada_merge_3, ac, how = 'outer',
                       left_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'],
                       right_on=['Survey Year', 'UNITID', 'Institution Name', 'Classification Name'])

I then created an additional column 'Total FB Coaches' by adding together the 'HC Total' and 'AC Total' columns.

In [20]:
eada_final['Total FB Coaches'] = eada_final['HC Total'] + eada_final['AC Total']

In [21]:
eada_final

Survey Year  UNITID                      Institution Name  \
0             2005  222178          Abilene Christian University   
1             2005  138558  Abraham Baldwin Agricultural College   
2             2005  126182                   Adams State College   
3             2005  188429                    Adelphi University   
4             2005  188438          Adirondack Community College   
...            ...     ...                                   ...   
29441         2019  181853                          York College   
29442         2019  217059          York College of Pennsylvania   
29443         2019  141361                  Young Harris College   
29444         2019  206695           Youngstown State University   
29445         2019  126119                          Yuba College   

                       Classification Name  Grand Total Revenue  \
0         NCAA Division II (with football)              5400385   
1                         NJCAA Division I               643297   
2         NCAA Division II (with football)              2781663   
3      NCAA Division II (without football)              4249797   
4                       NJCAA Division III               114840   
...                                    ...                  ...   
29441                     NAIA Division II              3805596   
29442   NCAA Division III without football              2857778   
29443    NCAA Division II without football              5751311   
29444                  NCAA Division I-FCS             16292051   
29445                                CCCAA               743969   

       Grand Total Expenses  Total Male Participation  \
0                   5074419                     207.0   
1                    643297                      58.0   
2                   2781663                     264.0   
3                   3556017                     145.0   
4                    111670                      62.0   
...                     ...                       ...   
29441               3805596                     166.0   
29442               2638512                     259.0   
29443               5751311                     191.0   
29444              16292051                     260.0   
29445                743969                     134.0   

       Total Female Participation  HC Total  AC Total  Total FB Coaches  
0                            89.0       1.0       7.0               8.0  
1                            40.0       NaN       NaN               NaN  
2                           110.0       1.0       6.0               7.0  
3                           105.0       NaN       NaN               NaN  
4                            30.0       NaN       NaN               NaN  
...                           ...       ...       ...               ...  
29441                       146.0       NaN       NaN               NaN  
29442                       221.0       NaN       NaN               NaN  
29443                       123.0       NaN       NaN               NaN  
29444                       187.0       1.0      10.0              11.0  
29445                        55.0       1.0       4.0               5.0  

[29446 rows x 11 columns]

---

## The Integrated Postsecondary Education Data System (IPEDS)

The last source used in this project is the [The Integrated Postsecondary Education Data System (IPEDS website](https://nces.ed.gov/ipeds/). I found this to be the trickiest website to deal with just given the sheer amount of data available. From the homepage, click on "Use The Data." From here, I scrolled over to "Survey Data" and then selected "Custom Data Files."

![img](../Images/IPEDS_data.png)

From here, the first step is to select the institutions that you want to include in your report. Again, given that I wanted to collect as much data as possible, I opted to select all institutions by selection "EZ Group" from the "By Groups" tab. From here you can select the box marked "All institutions" and then click "Search" to move on to variable selection.

![img](../Images/IPEDS_group.png)

![img](../Images/IPEDS_all_inst.png)

From the next page, I found the easiest way to collect the data I wanted was to search by variable name in the search bar. I was mostly interested in this database for student enrollment and retention numbers, so those were the first values I searched using keywords "enrollment" and "retention," respectively. I then searched for "average faculty salary" and "endowment" using the keywords "salary" and "endowment," respectively, to get some additional academic financial data. The exact variable names on the IPEDS site I used are as follows:

- Total Enrollment
- Full-time Enrollment
- Part-time Enrollment
- Full-time Retention Rate
- Part-time Retention Rate
- Average salary equated to 9-month contracts of full-time instructional faculty - all ranks
- Value of endowment assets at the end of the fiscal year
    - Note that the IPEDS website has separate categories for public and private institutions, so I had to select both variables for each year.

One aspect of collecting data from this website that is somewhat difficult is that if you want multiple years of data for each variable, you have to enter the same search variable for each year you want to download. Luckily the website allows you a maximum of 15 years worth of data at a time, so I was able to create a CSV file containing all 7 variables for each year between 2005–2019 (**Note that the 'year' for many variables on this website refers to the fall semester. In other words, to be consistent with my other data, I had to download data labeled '2004' for all variables except 'Endowment' to pertain to the '2005' data, given that the fall 2004 semester and spring 2005 semester equal one academic year. Endowmnet data was labeled by fiscal year, so for those variables I downloaded years 2005–2019.**). The only step after downloading these CSV files was merging them all together. There might have been an easier way to do this, but to format the data in the way I knew it would work I merged all 15 CSV files and then sorted them by year and institution name. I then downloaded this table as an Excel file and moved all of the columns to the left (you can paste and 'Skip Blanks' to avoid overwriting data) in order to eliminate all of the null values. I tried writing a function that would overwrite null values and move everything to the left of the dataframe in pandas, but there was a lot of inconsistency with which institutions had missing values for certain variables, so I decided to do this "by hand" to make sure the data was correct. Given that institutions are either public or private, I combined the two endowment columns into one 'Endowment' variable. 

![img](../Images/IPEDS_enrollment.png)

![img](../Images/IPEDS_download.png)

In [22]:
ipeds_2005 = pd.read_csv('../Data/IPEDS_2004.csv')
ipeds_2006 = pd.read_csv('../Data/IPEDS_2005.csv')
ipeds_2007 = pd.read_csv('../Data/IPEDS_2006.csv')
ipeds_2008 = pd.read_csv('../Data/IPEDS_2007.csv')
ipeds_2009 = pd.read_csv('../Data/IPEDS_2008.csv')
ipeds_2010 = pd.read_csv('../Data/IPEDS_2009.csv')
ipeds_2011 = pd.read_csv('../Data/IPEDS_2010.csv')
ipeds_2012 = pd.read_csv('../Data/IPEDS_2011.csv')
ipeds_2013 = pd.read_csv('../Data/IPEDS_2012.csv')
ipeds_2014 = pd.read_csv('../Data/IPEDS_2013.csv')
ipeds_2015 = pd.read_csv('../Data/IPEDS_2014.csv')
ipeds_2016 = pd.read_csv('../Data/IPEDS_2015.csv')
ipeds_2017 = pd.read_csv('../Data/IPEDS_2016.csv')
ipeds_2018 = pd.read_csv('../Data/IPEDS_2017.csv')
ipeds_2019 = pd.read_csv('../Data/IPEDS_2018.csv')

In [23]:
merge_1 = pd.merge(ipeds_2005, ipeds_2006, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_2 = pd.merge(merge_1, ipeds_2007, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_3 = pd.merge(merge_2, ipeds_2008, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_4 = pd.merge(merge_3, ipeds_2009, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_5 = pd.merge(merge_4, ipeds_2010, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_6 = pd.merge(merge_5, ipeds_2011, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_7 = pd.merge(merge_6, ipeds_2012, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_8 = pd.merge(merge_7, ipeds_2013, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_9 = pd.merge(merge_8, ipeds_2014, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_10 = pd.merge(merge_9, ipeds_2015, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_11 = pd.merge(merge_10, ipeds_2016, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

merge_12 = pd.merge(merge_11, ipeds_2017, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

ipeds_merged = pd.merge(merge_12, ipeds_2018, how='outer',
                  left_on=['unitid', 'institution name', 'year'],
                  right_on=['unitid', 'institution name', 'year'])

In [24]:
ipeds_merged.sort_values(by=['institution name', 'year'], inplace=True)
ipeds_merged

unitid                            institution name  year  \
6274   493868            A Better U Beauty Barber Academy  2004   
12714  493868            A Better U Beauty Barber Academy  2005   
19154  493868            A Better U Beauty Barber Academy  2006   
25594  493868            A Better U Beauty Barber Academy  2007   
32034  493868            A Better U Beauty Barber Academy  2008   
...       ...                                         ...   ...   
62787  446534  eClips School of Cosmetology and Barbering  2013   
69227  446534  eClips School of Cosmetology and Barbering  2014   
75667  446534  eClips School of Cosmetology and Barbering  2015   
82107  446534  eClips School of Cosmetology and Barbering  2016   
88547  446534  eClips School of Cosmetology and Barbering  2017   

       F0304_F2_RV.Value of endowment assets at the end of the fiscal year  \
6274                                                 NaN                     
12714                                                NaN                     
19154                                                NaN                     
25594                                                NaN                     
32034                                                NaN                     
...                                                  ...                     
62787                                                NaN                     
69227                                                NaN                     
75667                                                NaN                     
82107                                                NaN                     
88547                                                NaN                     

       F0304_F1A_RV.Value of endowment assets at the end of the fiscal year  \
6274                                                 NaN                      
12714                                                NaN                      
19154                                                NaN                      
25594                                                NaN                      
32034                                                NaN                      
...                                                  ...                      
62787                                                NaN                      
69227                                                NaN                      
75667                                                NaN                      
82107                                                NaN                      
88547                                                NaN                      

       DFR2004_RV.Total  enrollment  DFR2004_RV.Full-time enrollment  \
6274                            NaN                              NaN   
12714                           NaN                              NaN   
19154                           NaN                              NaN   
25594                           NaN                              NaN   
32034                           NaN                              NaN   
...                             ...                              ...   
62787                           NaN                              NaN   
69227                           NaN                              NaN   
75667                           NaN                              NaN   
82107                           NaN                              NaN   
88547                           NaN                              NaN   

       DFR2004_RV.Part-time enrollment  EF2004D_RV.Full-time retention rate  \
6274                               NaN                                  NaN   
12714                              NaN                                  NaN   
19154                              NaN                                  NaN   
25594                              NaN                                  NaN   
32034                              NaN                  

After merging and sorting the data, I eliminated schools that were not in the Knight Commission data above. I knew that the Knight Commission data would be vital to the project so I wanted to minimize the null values I would have to deal with in the summary notebook by trimming down the 90,160 rows in this merged data set.

In [25]:
# Creating a list of college names in the Knight data set
colleges = [x for x in knight['Data'].unique()]

In [26]:
# Dropping rows in the IPEDS data that do not match with the college names from the Knight Commission.
ipeds_final = ipeds_merged[ipeds_merged['institution name'].isin(colleges)]
ipeds_final

unitid              institution name  year  \
3253   222178  Abilene Christian University  2004   
9693   222178  Abilene Christian University  2005   
16133  222178  Abilene Christian University  2006   
22573  222178  Abilene Christian University  2007   
29013  222178  Abilene Christian University  2008   
...       ...                           ...   ...   
60671  206695   Youngstown State University  2013   
67111  206695   Youngstown State University  2014   
73551  206695   Youngstown State University  2015   
79991  206695   Youngstown State University  2016   
86431  206695   Youngstown State University  2017   

       F0304_F2_RV.Value of endowment assets at the end of the fiscal year  \
3253                                         158081120.0                     
9693                                                 NaN                     
16133                                                NaN                     
22573                                                NaN                     
29013                                                NaN                     
...                                                  ...                     
60671                                                NaN                     
67111                                                NaN                     
73551                                                NaN                     
79991                                                NaN                     
86431                                                NaN                     

       F0304_F1A_RV.Value of endowment assets at the end of the fiscal year  \
3253                                                 NaN                      
9693                                                 NaN                      
16133                                                NaN                      
22573                                                NaN                      
29013                                                NaN                      
...                                                  ...                      
60671                                                NaN                      
67111                                                NaN                      
73551                                                NaN                      
79991                                                NaN                      
86431                                                NaN                      

       DFR2004_RV.Total  enrollment  DFR2004_RV.Full-time enrollment  \
3253                         4761.0                           4273.0   
9693                            NaN                              NaN   
16133                           NaN                              NaN   
22573                           NaN                              NaN   
29013                           NaN                              NaN   
...                             ...                              ...   
60671                           NaN                              NaN   
67111                           NaN                              NaN   
73551                           NaN                              NaN   
79991                           NaN                              NaN   
86431                           NaN                              NaN   

       DFR2004_RV.Part-time enrollment  EF2004D_RV.Full-time retention rate  \
3253                             488.0                                 77.0   
9693                               NaN                                  NaN   
16133                              NaN                                  NaN   
22573                              NaN                                  NaN   
29013                              NaN                                  NaN   
...                                ...                                  ...   
60671                              NaN                            

As you can see, there are a lot of missing values in the merged data set. At this point, I saved the above data frame as an Excel file, then moved all of the colums to the left of the dataframe to erase the nulls, and renamed the column names for ease of interpretation. I added 1 to the 'year' column so that these values would match with the Knight and EADA data sets and ended up with the Excel file you see below. There are still missing values, but again these will be dealt with in the summary notebook, so I am not concerned about dealing with them here.

In [27]:
ipeds_final = pd.read_excel('../data/ipeds_final.xlsx', index_col=0)
ipeds_final

unitid             institution name  year  Total Enrollment  \
0     100654     Alabama A & M University  2005              6323   
1     100654     Alabama A & M University  2006              6182   
2     100654     Alabama A & M University  2007              6076   
3     100654     Alabama A & M University  2008              5706   
4     100654     Alabama A & M University  2009              5124   
...      ...                          ...   ...               ...   
2485  206695  Youngstown State University  2015             12503   
2486  206695  Youngstown State University  2016             12442   
2487  206695  Youngstown State University  2017             12643   
2488  206695  Youngstown State University  2018             12521   
2489  206695  Youngstown State University  2019             12607   

      Full-time Enrollment  Part-time Enrollment  Full-time Retention Rate  \
0                     5209                  1114                        64   
1                     5082                  1100                        69   
2                     5020                  1056                        70   
3                     4773                   933                        69   
4                     4330                   794                        68   
...                    ...                   ...                       ...   
2485                  9410                  3093                        70   
2486                  9267                  3175                        75   
2487                  9497                  3146                        75   
2488                  9564                  2957                        77   
2489                  9687                  2920                        76   

      Part-time Retention Rate  Average Faculty Salary    Endowment  
0                         45.0                   51089          NaN  
1                         85.0                   50908   17836152.0  
2                         46.0                   53544          NaN  
3                          0.0                   56522          NaN  
4                         33.0                   60575          NaN  
...                        ...                     ...          ...  
2485                      34.0                   72207  232153783.0  
2486                      47.0                   71982  225565166.0  
2487                      43.0                   73535  217669111.0  
2488                      58.0                   72465  241403045.0  
2489                      33.0                   74001  259291802.0  

[2490 rows x 10 columns]

For instructions on how to merge all three of these data sets, please see section 2.3 in the [summary notebook](https://github.com/Shoemaker703/college_retention/blob/main/Summary%20Notebook.ipynb) located on the main level of this repository.

I hope that this helped you better understand the process I went through for collecting the data used in this project. If you have any other questions, please feel free to reach out to me.